Compilation warning

Compilation fails when using the `-Werror=true` option.
`meson 1.1.0`
`gcc 13.1.1 "cc (GCC) 13.1.1 20230429"`

```
../librz/asm/arch/xap/dis.c: In function ‘decode_known’:
../librz/asm/arch/xap/dis.c:431:47: error: ‘snprintf’ output may be truncated before the last format character [-Werror=format-truncation=]
  431 |         snprintf(fmt, sizeof(fmt), "%s0x%%.%dX", sign, fmtsz);
      |                                               ^
../librz/asm/arch/xap/dis.c:431:9: note: ‘snprintf’ output 7 or more bytes (assuming 17) into a destination of size 16
  431 |         snprintf(fmt, sizeof(fmt), "%s0x%%.%dX", sign, fmtsz);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/librz/analysis/p/analysis_x86_cs.c
+++ b/librz/analysis/p/analysis_x86_cs.c
@@ -291,29 +291,29 @@ static char *getarg(struct Getarg *gop, int n, int set, char *setop, int sel, ut
 
 		if (index) {
 			if (scale > 1) {
-				snprintf(buf_, BUF_SZ, "%s%s,%d,*,", out, index, scale);
+				rz_strf(buf_, "%s%s,%d,*,", out, index, scale);
 			} else {
-				snprintf(buf_, BUF_SZ, "%s%s,", out, index);
+				rz_strf(buf_, "%s%s,", out, index);
 			}
 			strncpy(out, buf_, BUF_SZ);
 			component_count++;
 		}
 
 		if (base) {
-			snprintf(buf_, BUF_SZ, "%s%s,", out, base);
+			rz_strf(buf_, "%s%s,", out, base);
 			strncpy(out, buf_, BUF_SZ);
 			component_count++;
 		}
 
 		if (component_count > 1) {
 			if (component_count > 2) {
-				snprintf(buf_, BUF_SZ, "%s+,", out);
+				rz_strf(buf_, "%s+,", out);
 				strncpy(out, buf_, BUF_SZ);
 			}
 			if (disp < 0) {
-				snprintf(buf_, BUF_SZ, "%s-", out);
+				rz_strf(buf_, "%s-", out);
 			} else {
-				snprintf(buf_, BUF_SZ, "%s+", out);
+				rz_strf(buf_, "%s+", out);
 			}
 			strncpy(out, buf_, BUF_SZ);
 		} else {
@@ -327,13 +327,13 @@ static char *getarg(struct Getarg *gop, int n, int set, char *setop, int sel, ut
 		// set = 2 is reserved for lea, where the operand is a memory address,
 		// but the corresponding memory is not loaded.
 		if (set == 1) {
-			snprintf(buf_, BUF_SZ, "%s,%s=[%d]", out, setarg, op.size == 10 ? 8 : op.size);
+			rz_strf(buf_, "%s,%s=[%d]", out, setarg, op.size == 10 ? 8 : op.size);
 			strncpy(out, buf_, BUF_SZ);
 		} else if (set == 0) {
 			if (!*out) {
 				strcpy(out, "0");
 			}
-			snprintf(buf_, BUF_SZ, "%s,[%d]", out, op.size == 10 ? 8 : op.size);
+			rz_strf(buf_, "%s,[%d]", out, op.size == 10 ? 8 : op.size);
 			strncpy(out, buf_, BUF_SZ);
 		}
 		out[BUF_SZ - 1] = 0;

--- a/librz/asm/arch/xap/dis.c
+++ b/librz/asm/arch/xap/dis.c
@@ -428,7 +428,7 @@ static int decode_known(struct state *s, struct directive *d) {
 		fmtsz += 2;
 
 	// can be cleaned, no need to fmtsz
-	snprintf(fmt, sizeof(fmt), "%s0x%%.%dX", sign, fmtsz);
+	rz_strf(fmt, "%s0x%%.%dX", sign, fmtsz);
 	snprintf(tmp, sizeof(tmp), fmt, d->d_operand);
 	strcat(d->d_asm, tmp);
 

--- a/librz/debug/p/native/linux/linux_debug.h
+++ b/librz/debug/p/native/linux/linux_debug.h
@@ -173,7 +173,7 @@ int linux_reg_write(RzDebug *dbg, int type, const ut8 *buf, int size);
 RzList /*<RzDebugDesc *>*/ *linux_desc_list(int pid);
 bool linux_stop_threads(RzDebug *dbg, int except);
 int linux_handle_signals(RzDebug *dbg, int tid);
-int linux_dbg_wait(RzDebug *dbg, int pid);
+RzDebugReasonType linux_dbg_wait(RzDebug *dbg, int pid);
 char *linux_reg_profile(RzDebug *dbg);
 int match_pid(const void *pid_o, const void *th_o);
 

